src/plugins/release-merge-purifier/index.js

```diff
@@ -0,0 +1,239 @@
+/**
+ * @fileoverview Adds fail status to all non-semver pr's after release
+ * @author Gyandeep Singh
+ */
+
+"use strict";
+
+const semver = require("semver");
+
+const TAG_REGEX = /^(?:Build|Chore|Docs|Fix):/;
+
+/**
+ * Apply different checks on the commit message to see if its ok for a patch release
+ * @param {string} message - commit message
+ * @returns {boolean} `true` if the commit message is valid for patch release
+ * @private
+ */
+function isMessageValidForPatchRelease(message) {
+    return TAG_REGEX.test(message);
+}
+
+/**
+ * Get the correct commit message for a PR
+ * @param {Object} allCommits - Collection of all commit objects
+ * @param {Object} context - context object
+ * @returns {string} Commit message
+ * @private
+ */
+function getCommitMessageForPR(allCommits, context) {
+    return allCommits.data.length === 1
+        ? allCommits.data[0].commit.message
+        : context.payload.pull_request.title;
+}
+
+/**
+ * Check if the version is atleast minor
+ * @param {string} version - version to check
+ * @returns {boolean} true if its minor version
+ * @private
+ */
+function atLeastMinorVersion(version) {
+    return semver.patch(semver.clean(version)) === 0;
+}
+
+/**
+ * Get the sha value from the latest commit
+ * @param {Object} allCommits - all commit data
+ * @returns {string} latest commit sha
+ * @private
+ */
+function pluckLatestCommitSha(allCommits) {
+    return allCommits.data[allCommits.data.length - 1].sha;
+}
+
+/**
+ * Gets all the open PR
+ * @param {Object} context - context object
+ * @returns {Array<Object>} collection of release objects
+ * @private
+ */
+function getAllOpenPRs(context) {
+    return context.github.pullRequests.getAll(
+        context.issue({
+            state: "open"
+        })
+    );
+}
+
+/**
+ * Create status on the PR
+ * @param {Object} context - probot context object
+ * @param {string} state - state can be either success or failure
+ * @param {string} sha - sha for the commit
+ * @returns {Promise} Resolves when the status is created on the PR
+ * @private
+ */
+function createStatusOnPR(context, state, sha) {
```

Comment:
Nitpick: I think it would be better to accept an object argument here:

```js
function createStatusOnPR({ context, state, sha }) {
```

That way, calls to this method will be more readable because the arguments are annotated (e.g. it's clear what the `"success" string is used for).

```js
createStatusOnPR({ context, state: "success", sha });
```

Reply:
I like ur suggestions. will make changes